pkg/readiness/ready_tracker.go

```diff
@@ -131,16 +131,30 @@ func (t *Tracker) ForData(gvk schema.GroupVersionKind) Expectations {
 	return t.data.Get(gvk)
 }
 
-// CancelTemplate stops expecting the provided ConstraintTemplate and associated Constraints.
-func (t *Tracker) CancelTemplate(ct *templates.ConstraintTemplate) {
-	log.V(1).Info("cancel tracking for template", "namespace", ct.GetNamespace(), "name", ct.GetName())
-	t.templates.CancelExpect(ct)
+func (t *Tracker) templateCleanup(ct *templates.ConstraintTemplate) {
 	gvk := constraintGVK(ct)
 	t.constraints.Remove(gvk)
 	<-t.ready // constraintTrackers are setup in Run()
 	t.constraintTrackers.Cancel(gvk.String())
 }
 
+// CancelTemplate stops expecting the provided ConstraintTemplate and associated Constraints.
+func (t *Tracker) CancelTemplate(ct *templates.ConstraintTemplate) {
```

Comment:
Since you are replacing `tracker.CancelTemplate` with `tracker.TryCancelTemplate` in constrainttemplate_controller.go, where do we still call this function?

Reply:
It might make sense to have this function regardless of whether it's called, just for parity with the other trackers.

Reply:
It is still called on [line 481](https://github.com/julianKatz/gatekeeper/blob/177367246-1063-need-readiness-tracker-retries-for-canceltemplate-as-well/pkg/controller/constrainttemplate/constrainttemplate_controller.go#L481) of `constrainttemplate_controller.go`

If a constraint template delete event is received, this function is called.